src/analysis/Environments/Environments.ts

```diff
@@ -139,13 +149,15 @@ export const Environments = (props: EnvironmentsProps): ReactNode => {
     const endTimeForLeoCallsEpochMs = Date.now();
 
     const leoCallTimeTotalMs = endTimeForLeoCallsEpochMs - startTimeForLeoCallsEpochMs;
-    metrics.captureEvent(Events.cloudEnvironmentDetailsLoad, {
-      leoCallTimeMs: leoCallTimeTotalMs,
-      totalCallTimeMs: leoCallTimeTotalMs,
-      runtimes: newRuntimes.length,
-      disks: newDisks.length,
-      apps: newApps.length,
-    });
+    if (onEvent) {
+      onEvent('onDataRefresh', {
```

Comment:
See previous comment.

```suggestion
      onEvent('dataRefresh', {
```

Reply:
Nit: would prefer if this string was defined somewhere instead of included as a literal in its various usages. Perhaps the new `runtime-common-text` file?

Reply:
The event names are defined in the `EnvironmentsEvents` interface in this file.

Reply:
I see, but it is a function name there. Does that interface and the typing of `onEvent?: KeyedEventHandler<EnvironmentsEvents>;` provide typescript safety on the string usage here? Just trying to ensure "compiler" safety on a typo in the string literal elsewhere

Reply:
Yes. `onEvent` is typed as `KeyedEventHandler<EnvironmentsEvents>`, so its first argument must be a key of `EnvironmentsEvents`.

Reply:
yes, as Nick said, the type safety is locked in... only defined event key names will work.